pkg/admissioncontroller/cloudprivateipconfig/cloudprivateipconfig_admission_controller.go

```diff
@@ -0,0 +1,55 @@
+package admissioncontroller
+
+import (
+	"context"
+	"fmt"
+
+	cloudnetworkv1 "github.com/openshift/api/cloudnetwork/v1"
+	cloudnetworkclientset "github.com/openshift/client-go/cloudnetwork/clientset/versioned"
+	"github.com/openshift/cloud-network-config-controller/pkg/admissioncontroller"
+
+	admissionv1 "k8s.io/api/admission/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/klog/v2"
+)
+
+var (
+	CloudPrivateIPConfigResource = metav1.GroupVersionResource{
+		Version:  cloudnetworkv1.SchemeGroupVersion.Version,
+		Group:    cloudnetworkv1.SchemeGroupVersion.Group,
+		Resource: "cloudprivateipconfigs",
+	}
+)
+
+type CloudPrivateIPConfigAdmissionController struct {
+	admissioncontroller.AdmissionController
+}
+
+func NewCloudPrivateIPConfigAdmissionController(
+	cloudNetworkClient *cloudnetworkclientset.Clientset,
+	kubeClient *kubernetes.Clientset) *CloudPrivateIPConfigAdmissionController {
+	return &CloudPrivateIPConfigAdmissionController{
+		AdmissionController: admissioncontroller.AdmissionController{
+			CloudNetworkClient: cloudNetworkClient,
+			KubeClient:         kubeClient,
+		},
+	}
+}
+
+// AdmissionFunc is the implementation of AdmissionControllerIntf for the
+// CloudPrivateIPConfig resource. This controller admission criteria is: the
+// node specified in .spec.node must exist
```

Comment:
Why do we need to enforce that? Why not just have the config controller set a failed condition on it? The config controller needs to be able to deal with the existence of configs for non-existent nodes anyway, due to race conditions, restarts, etc.

And CloudPrivateIPConfigs should only be created by openshift-sdn/ovn-kubernetes anyway, not directly by the user, so we can just ensure that they only generate correct objects.

I don't think there's enough benefit to this admission controller to make it worth running it and maintaining the code.

Reply:
yeah, cluster-admin is always supposed to be able to create/edit any object

Reply:
OK, I am dropping the admission controller.